runtime/lua/vim/deepcopy.lua

```diff
@@ -0,0 +1,17 @@
+local function deepcopy(orig)
```

Comment:
this would be something that could live directly in `src/nvim/lua/vim.lua`. And I would maybe take the one from `test/helpers.lua` instead.